app/scripts/lib/fxa-client.js

```diff
@@ -546,23 +544,6 @@ define(function (require, exports, module) {
           // /recovery_email/status returns `emailVerified` and
           // `sessionVerified`, we don't want those.
           return _.pick(response, 'email', 'verified');
-        })
-        .fail(function (err) {
```

Comment:
I never really liked having this logic down in the fxa-client, even though I put it there. The check didn't seem to follow from the method name. It's not moved to account.js->waitForSessionVerification.